rosbag2_storage_default_plugins/src/rosbag2_storage_default_plugins/sqlite/sqlite_storage.cpp

```diff
@@ -477,12 +485,24 @@ rosbag2_storage::BagMetadata SqliteStorage::get_metadata()
 void SqliteStorage::set_filter(
   const rosbag2_storage::StorageFilter & storage_filter)
 {
+  // keep current start time and start row_id
+  // set topic filter and reset read statement for re-read
   storage_filter_ = storage_filter;
+  read_statement_ = nullptr;
 }
 
 void SqliteStorage::reset_filter()
 {
-  storage_filter_ = rosbag2_storage::StorageFilter();
+  set_filter(rosbag2_storage::StorageFilter());
+}
+
+void SqliteStorage::seek(const rcutils_time_point_value_t & timestamp)
+{
+  // reset row id to 0 and set start time to input
+  // keep topic filter and reset read statement for re-read
+  seek_row_id_ = 0;
+  seek_time_ = timestamp;
+  read_statement_ = nullptr;
```

Comment:
I would prefer to leverage heavy weighting operations by making new request to the DB here inside `seek` API rather than just resetting `read_statement_` pointer.

It will affect performance in some cases. 
Someone could reasonably assume that `seek` API really doing necessary heavy lifting preparations for retrieving next message with requesting timestamp.
As for the example if we would have bag file with 16 GB of data and will do `seek` and than `has_next` `read_next`. seek  operation will complete almost instantaneously although `has_next` will take substantial amount of time to reared this 16 GB in memory. For instance `play_next()` call will highly likely hiccup after the `seek` operation.

Reply:
To optimize performance please consider to iterate throw messages in current `message_result_` to try to find message with requested timestamp before doing new SQL request to the DB.

Reply:
After digging deeper in to the SQLite implementation I figured out that iterating through messages in current `message_result_` will not be faster than making new request to the database since when iterating throw `message_result_` we are actually doing new SQL request to the DB for the next row. 
I was thinking that `read_statement_->execute_query<..>` doing actual SQL request to the DB and caching it in the memory, but this is not true.